File: packages/engine/src/cpus/6502/instructions/dey.js
import { CycleType } from '..'

/**
 * DEcrement Y
 *
 * Y,Z,N = Y-1
 *
 * Affects Flags: N Z
 *
 * @example
 * MODE          SYNTAX        HEX LEN TIM
 * Implied       DEY           $88  1   2
 */
export function* dey() {
  if (--this.y < 0) this.y = 0xff
  this.zero = this.y === 0
  this.negative = (this.y & 0x80) > 0
  yield CycleType.GET
}

File: packages/engine/src/cores/famibob/cpu/instructions/dey.zig
const std = @import("std");
const CPU = @import("../cpu.zig").CPU;
const Addressing = @import("../opcode.zig").Addressing;

// DEcrement Y
//
// Y,Z,N = Y-1
//
// Affects Flags: N Z
//
// MODE          SYNTAX        HEX LEN TIM
// Implied       DEY           $88  1   2

pub fn dey(self: *CPU, addressing: Addressing) void {
    _ = addressing;
    self.y -%= 1;
    self.z = self.y == 0;
    self.n = (self.y & 0x80) > 0;
}
